starlite/config.py

```diff
@@ -141,6 +142,20 @@ def dict(self, *args, **kwargs) -> Dict[str, Any]:  # type: ignore[no-untyped-de
         return super().dict(*args, **kwargs)
 
 
+class CSRFConfig(BaseModel):
+    """Class containing CSRF configuration."""
+
+    secret: str
+    cookie_name: str = "csrftoken"
+    cookie_path: str = "/"
+    header_name: str = "x-csrftoken"
+    cookie_secure: bool = False
+    cookie_httponly: bool = False
+    cookie_samesite: str = "Lax"
+    cookie_domain: Optional[str] = None
+    safe_methods: Set[str] = {"GET", "HEAD", "OPTIONS", "TRACE"}
```

Comment:
Better to type as: `Set[literal["GET","POST", "PUT", "PATCH", "DELETE", "HEAD", "OPTIONS", "TRACE"]]`. Also see `starlite.types.Method`, you might want to update that type and use it instead.

Reply:
Good point, done!